### ISSUE TITLE
save bug

### ISSUE THREAD

**Issue by fayeshine** [fchollet/keras#6148]
```
In [3]: n.save('keras0')
---------------------------------------------------------------------------
RuntimeError                              Traceback (most recent call last)
<ipython-input-3-3059a54d88ac> in <module>()
----> 1 n.save('keras0')

/home/wenjian/anaconda3/lib/python3.5/site-packages/keras/engine/topology.py in save(self, filepath, overwrite)
   2414         """
   2415         from ..models import save_model
-> 2416         save_model(self, filepath, overwrite)
   2417 
   2418     def save_weights(self, filepath, overwrite=True):

/home/wenjian/anaconda3/lib/python3.5/site-packages/keras/models.py in save_model(model, filepath, overwrite)
    157                         name,
    158                         val.shape,
--> 159                         dtype=val.dtype)
    160                     if not val.shape:
    161                         # scalar

/home/wenjian/anaconda3/lib/python3.5/site-packages/h5py/_hl/group.py in create_dataset(self, name, shape, dtype, data, **kwds)
    106             dset = dataset.Dataset(dsid)
    107             if name is not None:
--> 108                 self[name] = dset
    109             return dset
    110 

h5py/_objects.pyx in h5py._objects.with_phil.wrapper (/home/ilan/minonda/conda-bld/work/h5py/_objects.c:2696)()

h5py/_objects.pyx in h5py._objects.with_phil.wrapper (/home/ilan/minonda/conda-bld/work/h5py/_objects.c:2654)()

/home/wenjian/anaconda3/lib/python3.5/site-packages/h5py/_hl/group.py in __setitem__(self, name, obj)
    275 
    276         if isinstance(obj, HLObject):
--> 277             h5o.link(obj.id, self.id, name, lcpl=lcpl, lapl=self._lapl)
    278 
    279         elif isinstance(obj, SoftLink):

h5py/_objects.pyx in h5py._objects.with_phil.wrapper (/home/ilan/minonda/conda-bld/work/h5py/_objects.c:2696)()

h5py/_objects.pyx in h5py._objects.with_phil.wrapper (/home/ilan/minonda/conda-bld/work/h5py/_objects.c:2654)()

h5py/h5o.pyx in h5py.h5o.link (/home/ilan/minonda/conda-bld/work/h5py/h5o.c:3743)()

RuntimeError: Unable to create link (Name already exists)

```

**Comment by xiaohk** [comment]
Same error here, with latest h5py and keras.

**Comment by yurii-andrieiev** [comment]
what is 'n'?

**Comment by fchollet** [comment]
> RuntimeError: Unable to create link (Name already exists)

Anyone has a code snippet to reproduce this error message?

**Comment by f11r** [comment]
Was fixed for me in 2.0.3
